### ISSUE TITLE
NCPA - Windows 2008 R2 EventViewer not working as expected

### ISSUE THREAD

**Issue by metabsd** [NagiosEnterprises/ncpa#383]
Hi,

I try to use NCPA to alert when a specific event id occur.

`https://servername:5693/api/logs?name=Application&logged_after=24h&severity=ERROR&event_id=10078`

NCPA return that result.

```
{
    "logs": [
        {
            "Application": []
        },
        "logs"
    ]
}
```

If I remove the eventid I receive that response.

`https://servername:5693/api/logs?name=Application&logged_after=24h&severity=ERROR`

```
{
    "logs": [
        {
            "Application": [
                {
                    "category": "1",
                    "severity": "ERROR",
                    "event_id": "17170698",
                    "application": "WDSIMGSRV",
                    "computer_name": "SERVERNAME",
                    "message": "An error occurred while refreshing settings. \r\n \r\n Error Information: 0x3A \r\n\r\n",
                    "time_generated": "09/15/17 06:55:02"
                },
                {
                    "category": "0",
                    "severity": "ERROR",
                    "event_id": "1511",
                    "application": "Microsoft-Windows-User Profiles Service",
                    "computer_name": "SERVERNAME",
                    "message": "Windows cannot find the local profile and is logging you on with a temporary profile. Changes you make to this profile will be lost when you log off.\r\n",
                    "time_generated": "09/15/17 00:05:00"
                },
                {
                    "category": "0",
                    "severity": "ERROR",
                    "event_id": "10500",
                    "application": "ServerView RAID",
                    "computer_name": "SERVERNAME",
                    "message": "Adapter LSI Embedded MegaRAID (0): State change on disk (0) from offline to failed \r\n",
                    "time_generated": "09/14/17 16:39:10"
                },
                {
                    "category": "0",
                    "severity": "ERROR",
                    "event_id": "10078",
                    "application": "ServerView RAID",
                    "computer_name": "SERVERNAME",
                    "message": "Adapter LSI Embedded MegaRAID (0): Logical drive 0 degraded \r\n",
                    "time_generated": "09/14/17 16:39:10"
                }
            ]
        },
        "logs"
    ]
}
```

**Comment by jomann09** [comment]
This may be related to the other bug fix for the windows event ids, let me test it out - the event_id should be filtered on after it has grabbed the logs so this is a bit odd. I was able to replicate it on mine also for any event_id < 1500 essentially. For whatever reason, some event_id's in the 8000s were working fine.

**Comment by jomann09** [comment]
Here is the event_id problem for next patch
https://github.com/NagiosEnterprises/ncpa/blob/master/agent/listener/windowslogs.py#L352

**Comment by senthilchokkalingam** [comment]
Regarding the event log monitoring, i tried the below query but it didnt filter correctly (also used event_id=4114). For one event (event id 1111), it filtered correctly but does not mark the event as critical.

[nagios@xi01 ~]$ /usr/local/nagios/libexec/check_ncpa.py -H IP -t 'Apptix123' -P 5693 -M 'logs' -q 'name=Application,severity=INFORMATION,logged_after=2h,eventid=4114' -w 0 -c 1 -l list
logs: [{u'Application': [{u'category': u'3', u'severity': u'INFORMATION', u'event_id': u'264224', u'application': u'MSExchange ADAccess', u'computer_name': u'server.lab.local', u'message': u'Process MAD.EXE (PID=3820). Exchange Active Directory Provider has discovered the following servers with the following characteristics:\r\n\r\n (Server name | Roles | Enabled | Reachability | Synchronized | GC capable | PDC | SACL right | Critical Data | Netlogon | OS Version)\r\n\r\nIn-site:\r\nlab.local\tCDG 1 7 7 1 0 1 1 7 1\r\nserver.lab.local\tCDG 1 7 7 1 0 1 1 7 1\r\n\r\nOut-of-site:\r\n\r\n\r\n\r\n', u'time_generated': u'10/09/17 14:26:58'}, {u'category': u'0', u'severity': u'INFORMATION', u'event_id': u'1003', u'application': u'Software Protection Platform Service', u'computer_name': u'server.lab.local', u'message': u'The Software Protection service has completed licensing status check.\r\nApplication Id=55c92734-d682-4d71-983e-d6ec3f16059f\r\nLicensing Status=\n1: 1d63ae00-7733-4e46-b0d0-ae0ab81d7638, 1, 0 [(0 [0xC004F014, 0, 0], [(?)(?)(?)(?)(?)(?)])(1 )(2 )]\n2: 4ae528f4-05c3-446e-90ea-a4fbd460b83a, 1, 0 [(0 [0xC004F014, 0, 0], [(?)(?)(?)(?)(?)(?)])(1 )(2 )]\n3: 


for the event id 1111, it can filter that log but still thats not marked as "Critical" even though i set critical threshold as 1.

[nagios@xi01 ~]$ /usr/local/nagios/libexec/check_ncpa.py -H IP -t 'Apptix123' -P 5693 -M 'logs' -q 'name=System,severity=ERROR,logged_after=2h,eventid=1111' -w 0 -c 1 -l list
logs: [{u'System': [{u'category': u'0', u'severity': u'ERROR', u'event_id': u'1111', u'application': u'UmrdpService', u'computer_name': u'server.lab.local', u'message': u'Driver Microsoft XPS Document Writer v4 required for printer Microsoft XPS Document Writer is unknown. Contact the administrator to install the driver before you log in again.\r\n', u'time_generated': u'10/09/17 14:26:32'}]}, u'logs']

Will this patch address address both of the above issues?  Also any time line on when the patch will be released? 

https://github.com/NagiosEnterprises/ncpa/blob/master/agent/listener/windowslogs.py#L352

**Comment by jomann09** [comment]
I'm sorry I phrased it poorly, the location is just so that I can find it again easier when I make the change. It will be added in the next release, which hopefully should be pretty soon. If I can get a patch out I will let you know, but since NCPA is compiled, it's much easier to wait for the next release.

**Comment by jomann09** [comment]
Fixed in 8cb5e42986cc2391417f2258af0a77ce6412f137
